docs/specs/editorconfig-in-the-compiler.md

```diff
@@ -0,0 +1,111 @@
+All-capital words are interpreted per [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+# Compiler Command Arguments
+
+The compiler *MUST* receive a new switch, `/editorconfig`, which accepts a single
+file which is an .editorconfig file that may apply to one or more files in the
+compilation. Multiple files are passed with multiple `/editorconfig` switches,
+the same way you can pass multiple references with /reference or multiple analyzers
+with /analyzer. There is no short form of the argument. Files passed with relative
```

Comment:
@jaredpar, for that to happen a user has to deviate from the normal `.editorconfig` pattern. The normal (and recommended) case is that the `.editorconfig` in the root of your repository defines the `root = true` attribute. This prevents any further processing up the chain.

So, if you have:
```
C:\Workspaces\.editorconfig
C:\Workspaces\Repo\.editorconfig		'root = true'
C:\Workspaces\Repo\SomeFile.cs
```

Then SomeFile.cs will use (and only use) C:\Workspaces\Repo\.editorconfig

If you have:
```
C:\Workspaces\.editorconfig
C:\Workspaces\Repo\.editorconfig		'root = true'
C:\Workspaces\Repo\Dir\.editorconfig
C:\Workspaces\Repo\Dir\SomeFile.cs
```

The SomeFile.cs will use C:\Workspaces\Repo\.editorconfig and C:\Workspaces\Repo\Dir\.editorconfig

Likewise if you have 
```
C:\Workspaces\.editorconfig
C:\Workspaces\Repo\.editorconfig		'root = true'
C:\Workspaces\Repo\Dir\.editorconfig	'root = true'
C:\Workspaces\Repo\Dir\SomeFile.cs
```

The SomeFile.cs will use (and only use) C:\Workspaces\Repo\Dir\.editorconfig


The case you mention can only happen if the user does:
C:\Workspaces\.editorconfig				'root = true'
C:\Workspaces\Repo\.editorconfig
C:\Workspaces\Repo\SomeFile.cs

In which case, SomeFile.cs will use C:\Workspaces\.editorconfig and C:\Workspaces\Repo\.editorconfig. It also means that moving the directory means the inheritance chain for the .editorconfig can change